### ISSUE TITLE
Allow disabling same package import check in UnusedImport

### ISSUE THREAD

**Issue by bendmorris** [adireddy/haxe-checkstyle#336]
An import of a file which defines multiple types will trigger an "UnusedImport" fail, even if some of those types are used. For example, I have haxepunk/Signal.hx which defines several types:

```
class Signal<T> {}
class Signal0 extends Signal<Void->Void> {}
class Signal1<T> extends Signal<T->Void> {}
```

If I `import haxepunk.Signal` to use Signal0 and Signal1, checkstyle complains: "Detected import haxepunk.Signal from same package haxepunk." Despite the fact that it's in the same package, I still have to import it to access types that aren't named Signal. (If I import each private type individually with `haxepunk.Signal.Signal0` it doesn't complain, but that's less than ideal.) This leads to a lot of false positives.

Can we either not warn in this case, and/or add an option to disable the same package import part of the check?

**Comment by AlexHaxe** [comment]
Hi, you could set `moduleTypeMap` or `ignoreModules` properties to either map types to a module or to completely ignore modules (= no checkstyle warnings). 
You can find a sample here: https://haxecheckstyle.github.io/docs/haxe-checkstyle/unusedimport.html
